From f80db3d5362895759eb1921b67cc4fb224c78def Mon Sep 17 00:00:00 2001
From: Tejun Heo <tj@kernel.org>
Date: Thu, 2 May 2024 10:51:16 -1000
Subject: [PATCH] rd-utils: Improve VERGEN_GIT_SHA handling

vergen generates VERGEN_IDEMPOTENT_OUTPUT when built out of tree. Let's
handle that.
---
 src/lib.rs | 14 +++++++++-----
 1 file changed, 9 insertions(+), 5 deletions(-)

diff --git a/src/lib.rs b/src/lib.rs
index f6841e3..4d1bb2a 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -48,15 +48,19 @@ pub const MSEC: f64 = 1.0 / 1000.0;
 pub const READ: usize = 0;
 pub const WRITE: usize = 1;
 
+#[rustfmt::skip]
 lazy_static::lazy_static! {
     static ref GIT_VERSION: String = {
         let mut ver = String::new();
-        if let Some(v) = option_env!("VERGEN_GIT_SHA") {
-            ver += "g";
-            ver += v;
-            if let Some("true") = option_env!("VERGEN_GIT_DIRTY") {
-                ver += "-dirty";
+	match option_env!("VERGEN_GIT_SHA") {
+            Some(v) if v != "VERGEN_IDEMPOTENT_OUTPUT" => {
+		ver += "g";
+		ver += v;
+		if let Some("true") = option_env!("VERGEN_GIT_DIRTY") {
+                    ver += "-dirty";
+		}
             }
+	    _ => {}
         }
         ver
     };
-- 
2.42.4

